### PR TITLE
[TM-5389] Projected Vacancy Integration (Edit)

### PR DESCRIPTION
**Pull request by jlam-mpc**
Updates
- Integrated GET language offset values for PV card
- Integrated GET metadata for PV card (created date, etc.)
- Integrated EDIT PV record
  - Used for card "save" button
  - Used for mass edit of include/exclude indicator when adding to proposed cycle
- Integrated EDIT PV language offset
- Integrated EDIT PV capsule description
- Fixed bug where array of included PVs were incorrect
- Refactored
  - Renamed fields
  - Reduced props in card

Pending:
- Display "None Listed" or "-" for empty values in card
- Display OBC Url
- Display proper language string in card

Dual Merge: 
- [API PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/936)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/437)

[TM-5389](https://metaphase.atlassian.net/browse/TM-5389)

[TM-5389]: https://metaphase.atlassian.net/browse/TM-5389?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by elizabeth-jimenez** [comment]
Update to use new Edit redux pattern, if possible

File: src/Components/ProjectedVacancyCard/ProjectedVacancyCard.jsx [hunk 1]
```diff
@@ -18,30 +18,53 @@ import TabbedCard from 'Components/TabbedCard';
...
-  const languageOffsets =
-    selectOptions?.languageOffsets?.length ? selectOptions.languageOffsets : [];
+  const summerLanguageOffsets = selectOptions?.languageOffsets?.summer_language_offsets?.length
+    ? selectOptions.languageOffsets.summer_language_offsets : [];
+  const winterLanguageOffsets = selectOptions?.languageOffsets?.winter_language_offsets?.length
+    ? selectOptions.languageOffsets.winter_language_offsets : [];
+
+  useEffect(() => {
+    dispatch(projectedVacancyMetadata({ future_vacancy_seq_num: result?.future_vacancy_seq_num }));
+    dispatch(projectedVacancyLanguageOffsets({ positon_seq_num: result?.position_seq_num }));
```
Comment (elizabeth-jimenez) [inline src/Components/ProjectedVacancyCard/ProjectedVacancyCard.jsx:43]: not seeing this being captured over in the action

Comment (elizabeth-jimenez) [reply src/Components/ProjectedVacancyCard/ProjectedVacancyCard.jsx:43]: by this i mean the argument

Comment (elizabeth-jimenez) [reply src/Components/ProjectedVacancyCard/ProjectedVacancyCard.jsx:43]: comment applies to both `projectedVacancyMetadata` and `projectedVacancyLanguageOffsets`

Comment (elizabeth-jimenez) [reply src/Components/ProjectedVacancyCard/ProjectedVacancyCard.jsx:43]: i wonder how the cancel is going to play with multiple calls tho 🤔

**Comment by elizabeth-jimenez** [comment]
I wonder if we should break out the   `const isLoading = filtersLoading || positionsLoading || languageOffsetsLoading;` to be filter agnostic and track the filter loading separately; that way the whole page doesn't blink on selection of a filter

**Comment by elizabeth-jimenez** [comment]
image
👀

File: src/Components/ProjectedVacancyCard/ProjectedVacancyCard.jsx [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+import { useDispatch, useSelector } from 'react-redux';
 import FA from 'react-fontawesome';
 import Linkify from 'react-linkify';
 import DatePicker from 'react-datepicker';
 import TextareaAutosize from 'react-textarea-autosize';
 import PropTypes from 'prop-types';
 import { useDidMountEffect } from 'hooks';
 import { projectedVacancyEdit } from 'actions/projectedVacancy';
-import { getDifferentials } from 'utilities';
+import { formatDateFromStr, getDifferentials } from 'utilities';
```
Comment (elizabeth-jimenez) [inline src/Components/ProjectedVacancyCard/ProjectedVacancyCard.jsx:10]:
```suggestion
import { formatDate, getDifferentials } from 'utilities';
```

**Comment by elizabeth-jimenez** [comment]
image
to Languages and maybe not show if they don't have?

**Comment by elizabeth-jimenez** [comment]
> image to Languages and maybe not show if they don't have?

i created a [function](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2879) we can use for this :)

File: src/actions/projectedVacancy.js [hunk 3]
```diff
@@ -228,92 +229,235 @@ export function projectedVacancyEdit(id, data) {
...
+          ));
+          dispatch(projectedVacancyEditLangOffsetsLoading(false));
+        });
+      })
+      .catch((err) => {
+        if (err?.message === 'cancel') {
+          batch(() => {
+            dispatch(projectedVacancyEditLangOffsetsLoading(true));
+            dispatch(projectedVacancyEditLangOffsetsErrored(false));
+          });
```
Comment (elizabeth-jimenez) [inline src/actions/projectedVacancy.js:289]: slightly new cancel pattern removes this, as it is redundant with the one on L268, 269 - checkout Publishable Positions for new pattern

**Review by elizabeth-jimenez — CHANGES_REQUESTED** [review]
1. break out loading for filters - optional
2. Date Formatting
3. Incumbent TED discrepancy
4. Language to plural
5. Languages formatting. Check function in new PR
6. Slight updates to Cancel pattern

**Comment by scott062** [comment]
Nice job Jeanie - I think there's a lot more requirement, as you noted, that we need to get clarification on, but I think this is at a workable stopping point and FE and BE are interacting nicely.

**Review by elizabeth-jimenez — APPROVED** [review]

**Pull request merged**